Bug: ng test fails to run because of ng-mocks error after upgrading to typescript 4.8.2

### Description of the bug

With devDependency on ng-mocks 14.2.0, after upgrading to typescript 4.8.2, `ng test` fails to run with the following error:

```
Error: node_modules/ng-mocks/index.d.ts:1447:112 - error TS2344: Type 'C' does not satisfy the constraint 'Record<string | number | symbol, any>'.

1447 export interface MockedComponentFixture<C = Record<string | number | symbol, any >, F = DefaultRenderComponent<C>> extends ComponentFixture<F> {
                                                                                                                    ~

  node_modules/ng-mocks/index.d.ts:1447:41
    1447 export interface MockedComponentFixture<C = Record<string | number | symbol, any >, F = DefaultRenderComponent<C>> extends ComponentFixture<F> {
                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    This type parameter might need an `extends Record<string | number | symbol, any>` constraint.
```

Everything works fine just downgrading typescript version to 4.7.4

--- a/libs/ng-mocks/src/lib/common/func.is-jest-mock.spec.ts
+++ b/libs/ng-mocks/src/lib/common/func.is-jest-mock.spec.ts
@@ -26,7 +26,7 @@ describe('func.is-jest-mock', () => {
   });
 
   it('handles objects', () => {
-    const test: Record<any, any> = {};
+    const test: Record<keyof any, any> = {};
     expect(funcIsJestMock(test)).toEqual(false);
 
     test._isMockFunction = true;

--- a/libs/ng-mocks/src/lib/mock-builder/mock-builder.performance.ts
+++ b/libs/ng-mocks/src/lib/mock-builder/mock-builder.performance.ts
@@ -78,7 +78,7 @@ export class MockBuilderPerformance extends MockBuilderPromise {
     return config;
   }
 
-  private equalsTo(prototype: Record<any, any>): boolean {
+  private equalsTo(prototype: Record<keyof any, any>): boolean {
     for (const key of ['beforeCC', 'keepDef', 'replaceDef', 'excludeDef', 'mockDef']) {
       if (!areEqualSets((this as any)[key], prototype[key])) {
         return false;

--- a/libs/ng-mocks/src/lib/mock-component/mock-component.ts
+++ b/libs/ng-mocks/src/lib/mock-component/mock-component.ts
@@ -58,7 +58,7 @@ const mixRenderReorderViews = (
   }
 };
 
-const mixRenderApplyContext = (view: EmbeddedViewRef<any>, context: Record<any, any>): void => {
+const mixRenderApplyContext = (view: EmbeddedViewRef<any>, context: Record<keyof any, any>): void => {
   for (const contextKey of Object.keys(view.context)) {
     view.context[contextKey] = undefined;
   }
@@ -74,7 +74,7 @@ const mixRenderHandleViews = (
   templates: any[],
   views: Array<EmbeddedViewRef<any>>,
   indices: undefined | number[],
-  context: Record<any, any>,
+  context: Record<keyof any, any>,
 ): number => {
   let index = -1;
 

--- a/libs/ng-mocks/src/lib/mock-render/types.ts
+++ b/libs/ng-mocks/src/lib/mock-render/types.ts
@@ -75,6 +75,6 @@ export interface MockedComponentFixture<C = any, F = DefaultRenderComponent<C>> 
  *
  * @see https://ng-mocks.sudo.eu/api/MockRender
  */
-export type DefaultRenderComponent<MComponent extends Record<keyof any, any>> = {
+export type DefaultRenderComponent<MComponent> = {
   [K in keyof MComponent]: MComponent[K];
 };

--- a/libs/ng-mocks/src/lib/resolve/collect-declarations.ts
+++ b/libs/ng-mocks/src/lib/resolve/collect-declarations.ts
@@ -32,7 +32,7 @@ const pushDecorator = (decorators: string[], decorator: string): void => {
   }
 };
 
-const getAllKeys = <T>(instance: T): Array<keyof T> => {
+const getAllKeys = <T extends Record<keyof any, any>>(instance: T): Array<keyof T> => {
   const props: string[] = [];
   for (const key of Object.keys(instance)) {
     props.push(key);

